### PR TITLE
IEP-1043: Allow the user to configure IDF_TOOLS_PATH

### PR DESCRIPTION
**Pull request by alirana01**
## Description

A new setting in Espressif preference page is added that can allow the user to configure the idf tools path for installation.

Fixes # ([IEP-1043](https://jira.espressif.com:8443/browse/IEP-1043))

## Type of change

Please delete options that are not relevant.
- New feature (non-breaking change which adds functionality)

## How has this been tested?

Configure the property in the Preferences page to something else and install tools they should be installed and builds and basic operations should be working fine.

image

**Test Configuration**:
* ESP-IDF Version: any
* OS (Windows,Linux and macOS): all

## Checklist
- [x] PR Self Reviewed
- [x] Applied Code formatting
- [ ] Verified on all platforms - Windows,Linux and macOS

## Summary by CodeRabbit

- **Refactor**
  - Replaced usage of `System.getenv()` with `IDFUtil.getSystemEnv()` across various components for consistent environment variable handling.

- **User Interface**
  - Updated Espressif Preferences Page with a new IDF tools path setting, allowing users to browse and select a directory for IDF tools.
  
- **New Features**
  - Added new string constants for IDF tools installation directory selection in preferences.

- **Bug Fixes**
  - Ensured environment variables are resolved based on the platform, improving compatibility and reliability.

### CODE REVIEW

**Comment by AndriiFilippov** [comment]
@alirana01 hi ! 
Tested under: 
OS - Windows 10 
ESP-IDF: v5.2 

My previous tools were installed in the /TESTTOOL/Espressif folder. I have set this folder as a user variable.
image

Using this PR I have configured new IDF with next(default) tools folder:
image

Once tools are installed, I do see it still execute python from `/TESTTOOL/Espressif` folder
image


I feel like it should use the new tools folder instead. 
Is it correct ?

**Comment by kolipakakondal** [comment]
Hi @AndriiFilippov Please verify this

File: bundles/com.espressif.idf.core/src/com/espressif/idf/core/IDFCorePreferenceConstants.java [hunk 1]
```diff
@@ -31,6 +32,10 @@ public class IDFCorePreferenceConstants
 	public static final String IDF_GITHUB_ASSETS_DEFAULT = "dl.espressif.com/github_assets"; //$NON-NLS-1$
 	public static final String PIP_EXTRA_INDEX_URL = "PIP_EXTRA_INDEX_URL"; //$NON-NLS-1$
 	public static final String PIP_EXTRA_INDEX_URL_DEFAULT = "https://dl.espressif.com/pypi"; //$NON-NLS-1$
+	public static final String IDF_TOOLS_PATH = "IDF_TOOLS_PATH"; //$NON-NLS-1$
+	public static final String IDF_TOOLS_PATH_DEFAULT = Platform.getOS().equals(Platform.OS_WIN32)
+			? "%USER_PROFILE\\.espressif" //$NON-NLS-1$
+			: "HOME/.espressif"; //$NON-NLS-1$
```
[the remark this answers is not in the document]
Comment (alirana01) [reply bundles/com.espressif.idf.core/src/com/espressif/idf/core/IDFCorePreferenceConstants.java]: Thanks for pointing out fixed

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/preferences/EspresssifPreferencesPage.java [hunk 2]
```diff
@@ -100,16 +102,29 @@ private void addToolsInstallationSettings(Composite mainComposite)
 		gitAssetsText.setLayoutData(gitTextGridData);
 		pythonWheelText.setLayoutData(pythonTextGridData);
 
+		Label idfToolsPathLabel = new Label(toolsInstallationGroup, SWT.NONE);
+		idfToolsPathLabel.setText(Messages.EspressifPreferencesPage_EspIdfToolsInstallationDirectoryLabel);
+		idfToolsPathText = new Text(toolsInstallationGroup, SWT.SINGLE | SWT.NONE);
+		GridData idfToolsPathTextGridData = new GridData(SWT.FILL, SWT.CENTER, true, false);
+		idfToolsPathTextGridData.widthHint = 200;
+		idfToolsPathText.setLayoutData(idfToolsPathTextGridData);
```
[the remark this answers is not in the document]
Comment (alirana01) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/preferences/EspresssifPreferencesPage.java:113]: added

**Comment by alirana01** [comment]
@AndriiFilippov please try once the builds are fine, the issue was that the process runner was not resolving the home directory vars automatically. Also the config now should show the resolved value for the user home directory

**Comment by AndriiFilippov** [comment]
@alirana01 hi !

now I could see `.espressif` folder created and tools installed there. 
But once I build project I could see it is still using user env path tools. 

see logs. 
[buildLOG.txt](https://github.com/user-attachments/files/16097443/buildLOG.txt)

**Comment by alirana01** [comment]
> @alirana01 hi !
> 
> it still takes some python dependencies from user env path during build process:
> 
> ```

@AndriiFilippov can you please try again with latest build

**Comment by AndriiFilippov** [comment]
@alirana01 hi ! 

Still referring to user tool path -> during "ESP-IDF: Project Full Clean" execution: 

```
ERROR: C:\Users\AndriiFilippov\tool1\espidf.constraints.v5.1.txt doesn't exist. Perhaps you've forgotten to run the install scripts. Please check the installation guide for more information.

ESP-IDF v5.1.4-474-gbae9a3a29f
```

**Comment by AndriiFilippov** [comment]
@alirana01 hi ! 

Windows 10 

affected (still refer to usr tool path) : 

- "Espressif" -> Product Information 
- ESP-IDF: Python Clean
- ESP-IDF: Project  Clean
- ESP-IDF: Project Full Clean
- ESP-IDF: Application Size Analyzer 
- ESP-IDF: Install New Component 
- Can't open sdkconfig files

**Comment by kolipakakondal** [comment]
Hi @alirana01 Could you address the affected functionality as well?

**Comment by alirana01** [comment]
@AndriiFilippov can you verify these things again please make sure to test on other platforms as well by setting the IDF_TOOLS_PATH in system env to some other value

**Review by AndriiFilippov — APPROVED** [review]

**Review by kolipakakondal — APPROVED** [review]

**Pull request merged**